### PR TITLE
Allow login with wrong SA pwd

### PR DESCRIPTION
**Pull request by mrajashree**
When AD server's SA password changes, the users get locked out of
rancher server upon logout. This is because during login, the bind
using SA creds fails and as a result the queries to get user's group
attributes aren't done. This commit ensures that user can still login
even if the SA bind fails, by returning user's groups' identities
formed based on DN instead of by querying AD for friendly name.

### CODE REVIEW

**Comment by mrajashree** [comment]
https://github.com/rancher/rancher/issues/9456

File: providers/ldap/ldap_client.go [hunk 1]
```diff
@@ -323,12 +324,36 @@ func (l *LClient) GetIdentity(distinguishedName string, scope string) (client.Id
 		return nilIdentity, fmt.Errorf("Error %v creating connection", err)
 	}
 	// Bind before query
+	// If service acc bind fails, and auth is on, return identity formed using DN
 	serviceAccountUsername := getUserExternalID(l.Config.ServiceAccountUsername, l.Config.LoginDomain)
 	err = lConn.Bind(serviceAccountUsername, l.Config.ServiceAccountPassword)
 	if err != nil {
-		return nilIdentity, fmt.Errorf("Error %v in ldap bind", err)
-	}
+		defer lConn.Close()
```
Comment (cjellick) [inline providers/ldap/ldap_client.go]: You defer close here and on line 357 right outside the if block. Can consolidate into one defer before the if block?

Comment (cjellick) [reply providers/ldap/ldap_client.go]: Is there any chance that lConn will be nil and lConn.Close() will panic?

File: providers/ldap/ldap_client.go [hunk 2]
```diff
@@ -323,12 +324,36 @@ func (l *LClient) GetIdentity(distinguishedName string, scope string) (client.Id
...
+		if ldap.IsErrorWithCode(err, ldap.LDAPResultInvalidCredentials) && !l.Enabled {
+			return nilIdentity, fmt.Errorf("Error in ldap bind: %v", err)
+		} else if ldap.IsErrorWithCode(err, ldap.LDAPResultInvalidCredentials) && l.Enabled {
+			if strings.EqualFold(c.UserScope, scope) {
```
Comment (cjellick) [inline providers/ldap/ldap_client.go]: Could this `if/else` block be condensed to
```
user = strings.EqualFold(c.UserScope, scope) 
```

Comment (mrajashree) [reply providers/ldap/ldap_client.go]: yes will do

**Comment by cjellick** [comment]
lgtm

**Pull request merged**